content/articles/email-parser-python/index.md

```diff
@@ -0,0 +1,170 @@
+SPAMS_DIR = os.path.join(SPAM_PATH, "spam_emails")
+
+# sorted filenames for the emails
+normal_filenames = [name for name in sorted(os.listdir(NORMALS_DIR))]
+spam_filenames = [name for name in sorted(os.listdir(SPAMS_DIR))]
+#load the emails
+def load_emails(is_spam, filename, spams_path=SPAM_PATH):
+  #if the argument is true for spam, load from the spam_emails directory and vice versa
+    directory = "spam_emails" if is_spam else "normal_emails"
+    #open as readable and in binary
+    with open(os.path.join(spams_path, directory, filename), "rb") as f:
+      #parse using the defaul line break(\n)
+        return email.parser.BytesParser(policy=email.policy.default).parse(f)
+#load
+normal_emails = [load_emails(is_spam=False, filename=name) for name in normal_filenames]
+spam_emails = [load_emails(is_spam=True, filename=name) for name in spam_filenames]
+```
+
+Here, we create a directory for the spam and normal emails and sort them.
+In the `load_emails()`, we open the appropriate directories as readable and in binary format, then we parse them using the `BytesParser` class.
+The `BytesParser` class contains an argument in the constructor called `policy`. It has the policy as default. Using `default` lets us parse the email using the `\n` line breaks.
+
+We then call the `load_emails()` method to load the emails.
+
+We can look at a sample of a normal email in the next cell.
+
+```python
+print(normal_emails[42].get_content().strip())
+```
+
+#### Viewing the structures of the emails
+
+In this part, we will look at how to get the structure of the emails and further look at the common types of structures for spam and normal emails.
+
+Paste this in the next cell.
+
+```python
+'''Getting the most common email structures'''
+def get_structures(email):
+  #if its plain text return text/plain
+    if isinstance(email, str):
+        return email
+    email_payload = email.get_email_payload()
+    #if the payload is a list then its probably a multipart
+    #return a multipart thereafter
+    if isinstance(email_payload, list):
+        return "multipart({})".format(", ".join([
+            get_structures(sub_email)
+            for sub_email in email_payload
+        ]))
+    else:
+        return email.get_content_type()
+#function for counting the types
+def type_counter(emails):
+    our_count = Counter()
+    for email in emails:
+        structure = get_structures(email)
+        our_count[structure] += 1
+    return our_count
+```
+
+In this code, we have two methods, `get_structures()` and `type_counter()`.
+
+For the `get_structures()`, we check for the structure. If it is a normal text email, we return `text/plain,` but if it's a multipart type of email, we return multipart, all the parts it contains. Note also that recursion is used if there are many email structures in the sub-emails of that email.
+For any other email structure we display it by return the emails content type(`get_content_type()`).
```

Comment:
You use for at the beginning of the sentences very frequently. Use better words to get in a sense of flow.